.github/workflows/test.yaml

```diff
@@ -23,22 +23,16 @@ jobs:
       fail-fast: false
       matrix:
         os: ['ubuntu-latest']
-        python-version: ['3.7']
+        python-version: ['3.7', '3.8', '3.9', '3.8', '3.9', '3.10']
 
     steps:
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - uses: actions/checkout@v3
 
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-
-      - uses: actions/cache@v2
-        with:
-          path: ${{ env.pythonLocation }}
-          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.cfg') }}
```

Comment:
Why was the caching removed here? This helps cut down CI time on setting up the python environment.

Reply:
its included in actions/setup_python now 3bdc2f7

Reply:
`key: ${{ env.pythonLocation }}-${{ hashFiles('setup.cfg') }}-${{ matrix.python-version }}`

Reply:
I think that means it caches the local pip package directory, whereas actions/cache@v2 caches the entire python environment. Much more efficient.

Reply:
i can add a cache-dependency-path so it is rebuild upon setup.cfg changes: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages:

cache-dependency-path: setup.cfg

Reply:
It's great that it can do that, but this does something fundamentally different than the previous caching we had, which is imo much more efficient and bypasses the need to keep the pip cache.

Reply:
You are correct, the direct caching saves all python stuff, instead of only the pip cache. 
the extra `${{ matrix.python-version }}` is not needed, that is already included in `${{ env.pythonLocation }}`

Reply:
I will fix it in #404